xonsh/environ.py

```diff
@@ -106,7 +106,6 @@ def to_debug(x):
     re.compile('\w*PATH$'): (is_env_path, str_to_env_path, env_path_to_str),
     'PATHEXT': (is_env_path, str_to_env_path, env_path_to_str),
     'RAISE_SUBPROC_ERROR': (is_bool, to_bool, bool_to_str),
-    'RIGHT_PROMPT': (is_string, ensure_string, ensure_string),
```

Comment:
Yeah, I don't think that checking the return type of a callable is good idea, since they can literally do anything :)